src/aoc2020/day1.clj

```diff
@@ -1,61 +1,102 @@
 (ns aoc2020.day1
-  (:require [clojure.java.io :as io]
-            [clojure.string :as str]))
-
-(defn file->seq
-  [filename]
-  (-> filename
-      (io/resource)
-      (slurp)
-      (str/split-lines)))
-;;Q1: find 2 numbers in the file that would sum up to be 2020, and find the product of these 2 numbers.
-
-;Approach 1: brute force
-;loop through the entries and use 2020 to minus each number, and loop through the rest of the vectors to find the
-;if the (2020 - num) exist. Time: O(n^2) space: O(1)
-
-
-;Approach 2: binary search
-;sort the vector in ascending order, and use 2 pointers pointing to head and end of the vector, and move pointers
-;based on the sum of 2 numbers. Time: O(log n) space: O(1)
-
-(defn binary-search
-  [data-set]
-  (->> map (fn [left-pointer
-                right-pointer]
-             ((if (== 2020 (+ (get data-set left-pointer)
-                              (get data-set right-pointer)))
-
-                )))
-       (range 0 (-> data-set count))
-       (range (-> data-set count) 0 -1))
-  )
+  (:require [aoc2020.util :refer [file->seq]]))
+
+(defn complement
+  [sum x]
+  (- sum x))
+
+
+(defn find-complement
+  [data-set sum]
+  (let [compl-exist?  (fn [num]
+                        (contains? data-set (complement sum num)))
+        result        (filterv compl-exist? data-set)]
+
+    (when (first result)
+      result)))
+
+(defn two-sum-product
+  [data-s sum]
+  (let [data-set     (into #{} data-s)
+        result       (find-complement data-set sum)
+        num1         (result 0)
+        num2         (result 1)]
```

Comment:
This was already addressed in a previous PR... using dynamically bound symbols as functions is dangerous and difficult to debug